hashing/src/lib.rs

```diff
@@ -107,14 +147,26 @@ impl Digest {
     /// l
     /// ```
     ///
+    /// Finally hashes the number of elements resulting hash. In the example above the final output
+    /// would be `hash_pair(u64_as_slice(6), l)`.
```

Comment:
The doc comment has been changed to reference the `Itertools::tree_fold1()`, so the example is no longer present.

Reply:
But isn't all the logic still the same? We used `Itertools::tree_fold1` because it's exactly equivalent with what we had implemented (and documented), so I'd rather keep the full example, including the diagram of the tree.

Reply:
@rafal-ch I'm with @afck. Please restore the documentation.  This is from the associated [CEP-41][1].  Along with restoring the documentation, I would suggest adding a reference to appropriate part of that spec here.

[1]: https://github.com/casper-network/ceps/blob/master/text/0041-extensible-block-body.md#:~:text=The%20third%20hashing%20primitive%20constructs%20a%20Merkle%20tree.%20It%20is%20intended%20for%20vectors%20where%20we%20do%20not%20care%20about%20order.%20The%20deploy_hashes%20field%20of%20BlockBody%20as%20previously%20discussed%20is%20an%20example.%20When%20handed%20an%20empty%20list%20the%20procedure%20returns%20the%20sentinel%20hash(%26%5B%5D).

Reply:
@Fraser999 @xcthulhu @afck 
As discussed on a call, I've re-added the documentation and used the corrected version of the graph:
```
1 2 4 5 8 9
│ │ │ │ │ │
└─3 └─6 └─10
  │   │   │
  └───7   │
      │   │
      └───11
```

Since the doc is back, I also applied the suggestion about `6_u64.to_le_bytes()`.

Please check the details in this commit: https://github.com/casper-network/casper-node/pull/2119/commits/b5e36e3c52aa6379a61044e6d75040b1863b0863